### PR TITLE
Reflectometry GUI: Add the ability to 'Fill Down' cells

### LINKED ISSUE #26025: ISIS Reflectometry GUI fixes and improvements
This epic contains remaining work and issues relating to the reflectometry GUI restructuring in #26016. Child issues have been created for all of the individual tasks and assigned milestones if relevant for higher priority items. Lower priority items do not yet have a milestone.

### PR DESCRIPTION
**Pull request by Pasarus**
**Description of work.**
I added a fill down button which functions like excel: https://support.office.com/en-us/article/fill-a-formula-down-into-adjacent-cells-041edfe2-05bc-40e6-b933-ef48c3f308c6 Except you do not have to only add them to adjacent cells and extra rows can be selected.

**To test:**
- Open MantidPlot and Workbench
- Open ISIS Reflectometry
- Create a group and multiple rows (atleast 5)
- Select the first row and add text to one of the columns
- Select the rows you wish to 
image
- Make sure you select the column you want to "FillDown"
- Click the FillDown button
- The text you entered should now have been put into each of the rows that were selected.

refs #26025 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by dtasev — CHANGES_REQUESTED** [review]
I found a few things a bit confusing:
- Selecting multiple columns and pressing fill down, only fills down the one that the selection ended on. This should copy the data from all selected columns. 
  - It could be done in a separate PR, but until then it should be stated (in release notes)
- Having the whole row highlighted, while having a cell selection, makes the cell selection invisible. I don't think the row highlight can be disabled without making other functionality in the GUI confusing - I wonder if a different palette can be used for selected cells
  - This could also be in a separate PR

File: docs/source/release/v4.1.0/reflectometry.rst [hunk 1]
```diff
@@ -26,6 +26,7 @@ New
 The Runs Table
 ^^^^^^^^^^^^^^
 
+- Added a "Fill Down" functionality similar to Excel
```
[the remark this answers is not in the document]
Comment (Pasarus) [reply docs/source/release/v4.1.0/reflectometry.rst]: We agreed offline that a change to the description of the feature is best.

**Review by dtasev — DISMISSED** [review]
After an offline discussion we agreed that the features above will only be done if the users request it

**Comment by NickDraper** [comment]
Can you add the Fill Down button image into the release notes.

**Comment by NickDraper** [comment]
I aborted the build to save on resources as this is an rst only change

**Pull request merged**